### PR TITLE
Minor fixes for rendering errors and search filter component not filtering out commute input on regular searches.

### PR DESCRIPTION
**Pull request by surgiie**
Fixes several rendering issues and commute input not cleared on filter urls.

See comments.

### CODE REVIEW

File: src/components/Search/AppSearchFilter.vue [hunk 1]
```diff
@@ -104,21 +104,24 @@ export default {
...
             displayedFilters: {},
         }
     },
     created() {
-
+        let value = null
         let filteredOptions = []
         this.givenOptions.forEach(option => {
-            if(this.input[this.name] != option.display){
+            value = this.optionHasSubmitValue(option) ? option.submit: option.display
```
Comment (surgiie) [inline src/components/Search/AppSearchFilter.vue:115]: This code just allows options to have a "submit" property on them to allow to specify what values should be submitting. We were already doing this before but i think i accidently removed it in some recent refactors. This allows the component to be used manually or for explicit filters not coming back from back end. Ex:

```html
     <AppSearchFilter
            name="r"
            display="Radius"
            :input="filteredInput"
            :options="[
                { display: 'Exact Location Only', submit: ''},
                { display: 'Within 25 Miles', submit: 25 },
                { display: 'Within 50 Miles', submit: 50 }
            ]"
        >
        </AppSearchFilter>
```

File: src/components/Search/Providers/AppGoogleTalentSearchProvider.vue [hunk 2]
```diff
@@ -17,6 +17,7 @@
             :appliedFilters="appliedFilters"
             :isCommuteSearch="isCommuteSearch"
             :getFilterOptions="getFilterOptions"
+            :filteredInput="getCurrentPayload()"
```
Comment (surgiie) [inline src/components/Search/Providers/AppGoogleTalentSearchProvider.vue:20]: Returns relevant data/input based on the search type being completed.

File: src/components/Search/Providers/mixins/provider.js [hunk 3]
```diff
@@ -160,7 +161,7 @@ export default  {
         },
         getFilterOptions(filter) {
             let key = blank(filter.key) ? filter.name : filter.key
-            return this.filters[key] || []
+            return (blank(this.filters[key]) || !Array.isArray(this.filters[key])) ? [] : this.filters[key]
```
Comment (surgiie) [inline src/components/Search/Providers/mixins/provider.js:164]: For some reason when you dont have mini kube running or you get a timeout `this.filters[key] || []` evaluates to empty sometimes other times Vue was returning a `mutator` function and that caused rendering issues with the prop validation on AppSearchFilter expecting an array for `options`

File: src/components/Search/Providers/mixins/provider.js [hunk 4]
```diff
@@ -60,7 +60,8 @@ export default  {
             return this.siteConfig.filters || []
         },
         sort() {
-            let sort = {}
+            let sort = { options: [], sortField: ()=>{}, by: null}
```
Comment (surgiie) [inline src/components/Search/Providers/mixins/provider.js:63]: Define some defaults so the html doesnt error out when no response comes back for sort meta.

File: src/components/Search/AppSearchFilter.vue (hunk 1 above)
Comment (jeremytubbs) [inline src/components/Search/AppSearchFilter.vue:115]: Can you add a space before the `:` of your ternary.

File: src/components/Search/AppSearchFilter.vue [hunk 5]
```diff
@@ -104,21 +104,24 @@ export default {
...
+            if(this.name == 'r'){
+                console.log(this.input, value)
```
Comment (jeremytubbs) [inline src/components/Search/AppSearchFilter.vue:117]: console.log?

File: src/components/Search/Providers/mixins/provider.js (hunk 4 above)
Comment (jeremytubbs) [reply src/components/Search/Providers/mixins/provider.js:63]: Please format this better with new lines

**Review by jeremytubbs — APPROVED** [review]

**Pull request merged**